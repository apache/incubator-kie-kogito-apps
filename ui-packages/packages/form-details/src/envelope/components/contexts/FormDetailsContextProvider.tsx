import React from 'react';
import RuntimeToolsFormDetailsContext, {
  FormDetailsContextImpl
} from './FormDetailsContext';

interface IOwnProps {}

const FormDetailsContextProvider: React.FC<IOwnProps> = ({ children }) => {
  return (
    <RuntimeToolsFormDetailsContext.Provider
      value={new FormDetailsContextImpl()}
    >
      {children}
    </RuntimeToolsFormDetailsContext.Provider>
  );
};

export default FormDetailsContextProvider;
