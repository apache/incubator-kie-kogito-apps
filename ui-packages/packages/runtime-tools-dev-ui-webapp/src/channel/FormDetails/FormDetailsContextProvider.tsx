import React from 'react';
import FormDetailsContext from './FormDetailsContext';
import { FormDetailsGatewayApiImpl } from './FormDetailsGatewayApi';

interface IOwnProps {
  children;
}

const FormDetailsContextProvider: React.FC<IOwnProps> = ({ children }) => {
  return (
    <FormDetailsContext.Provider value={new FormDetailsGatewayApiImpl()}>
      {children}
    </FormDetailsContext.Provider>
  );
};

export default FormDetailsContextProvider;
