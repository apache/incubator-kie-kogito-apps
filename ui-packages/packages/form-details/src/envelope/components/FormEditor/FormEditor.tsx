/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useImperativeHandle, useState } from 'react';
import { OUIAProps, componentOuiaProps } from '@kogito-apps/ouia-tools';
import {
  CodeEditor,
  CodeEditorControl,
  Language
} from '@patternfly/react-code-editor';
import {
  PlayIcon,
  RedoIcon,
  SaveIcon,
  UndoIcon
} from '@patternfly/react-icons';
import { Form } from '../../../api';
import { useFormDetailsContext } from '../contexts/FormDetailsContext';
import { ResizableContent } from '../FormDetails/FormDetails';
import '../styles.css';
export interface FormEditorProps {
  formType?: string;
  isSource?: boolean;
  isConfig?: boolean;
  formContent: Form;
  contentChange: Form;
  code: string;
  setFormContent: (formContent: Form) => void;
  setContentChange: (contentChange: Form) => void;
  saveFormContent: (formContent: Form) => void;
}

export const FormEditor = React.forwardRef<
  ResizableContent,
  FormEditorProps & OUIAProps
>(
  (
    {
      code,
      formType,
      formContent,
      setFormContent,
      saveFormContent,
      contentChange,
      setContentChange,
      isSource = false,
      isConfig = false,
      ouiaId,
      ouiaSafe
    },
    forwardedRef
  ) => {
    const appContext = useFormDetailsContext();

    const [monacoEditor, setMonacoEditor] = useState<any>();

    useImperativeHandle(
      forwardedRef,
      () => {
        return {
          doResize() {
            monacoEditor.layout();
          }
        };
      },
      [monacoEditor]
    );

    const getFormLanguage = (): Language => {
      if (isSource && formType) {
        switch (formType.toLowerCase()) {
          case 'tsx':
            return Language.typescript;
          case 'html':
            return Language.html;
        }
        /* istanbul ignore else */
      } else if (isConfig) {
        return Language.json;
      }
    };

    const editorDidMount = (editor, monaco): void => {
      if (isSource && formType.toLowerCase() === 'tsx') {
        monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
          jsx: 'react'
        });

        monaco.languages.typescript.typescriptDefaults.setDiagnosticsOptions({
          noSemanticValidation: false,
          noSyntaxValidation: false
        });
      }
      setMonacoEditor(editor);
    };

    const handleChange = (value): void => {
      if (Object.keys(formContent)[0].length > 0 && isSource) {
        const temp: Form = formContent;
        temp.source = value;
        setContentChange({ ...formContent, ...temp });
      } else {
        const temp: Form = formContent;
        temp.configuration['resources'] = JSON.parse(value);
        setContentChange({ ...formContent, ...temp });
      }
    };

    const onExecuteCode = (): void => {
      appContext.updateContent(contentChange);
      setFormContent(contentChange);
    };

    const onSaveForm = (): void => {
      saveFormContent(formContent);
    };

    const onUndoChanges = (): void => {
      if (monacoEditor !== null) {
        monacoEditor.focus();
        document.execCommand('undo');
      }
    };

    const onRedoChanges = (): void => {
      if (monacoEditor !== null) {
        monacoEditor.focus();
        document.execCommand('redo');
      }
    };

    const customControl = (
      <>
        <CodeEditorControl
          icon={<SaveIcon />}
          aria-label="Save form"
          toolTipText="Save form"
          onClick={() => onSaveForm()}
          isVisible={contentChange && Object.keys(contentChange)[0].length > 0}
        />
        <CodeEditorControl
          icon={<PlayIcon className="pf-global--primary-color--100" />}
          aria-label="Execute form"
          toolTipText="Execute form"
          onClick={onExecuteCode}
          isVisible={contentChange && Object.keys(contentChange)[0].length > 0}
        />
        <CodeEditorControl
          icon={<UndoIcon className="pf-global--primary-color--100" />}
          aria-label="Undo changes"
          toolTipText="Undo changes"
          onClick={onUndoChanges}
          isVisible={contentChange && Object.keys(contentChange)[0].length > 0}
        />
        <CodeEditorControl
          icon={<RedoIcon className="pf-global--primary-color--100" />}
          aria-label="Redo changes"
          toolTipText="Redo changes"
          onClick={onRedoChanges}
          isVisible={contentChange && Object.keys(contentChange)[0].length > 0}
        />
      </>
    );

    return (
      <div {...componentOuiaProps(ouiaId, 'form-view', ouiaSafe)}>
        <CodeEditor
          isCopyEnabled
          isDarkTheme={false}
          isLineNumbersVisible={true}
          isReadOnly={false}
          isMinimapVisible={false}
          isLanguageLabelVisible
          customControls={customControl}
          code={code}
          language={getFormLanguage()}
          height="800px"
          onEditorDidMount={editorDidMount}
          onChange={handleChange}
        />
      </div>
    );
  }
);

export default FormEditor;
