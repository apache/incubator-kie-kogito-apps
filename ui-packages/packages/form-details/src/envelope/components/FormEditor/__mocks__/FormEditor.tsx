import React from 'react';
import { FormEditorProps } from '../FormEditor';
import { ResizableContent } from '../../FormDetails/FormDetails';
import { OUIAProps } from '@kogito-apps/ouia-tools/dist/utils/OuiaUtils';

const MockedFormEditor = React.forwardRef<
  ResizableContent,
  FormEditorProps & OUIAProps
>((props, ref) => {
  return <></>;
});

export default MockedFormEditor;
