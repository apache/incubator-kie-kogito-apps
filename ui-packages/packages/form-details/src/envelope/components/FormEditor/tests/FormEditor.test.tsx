/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from 'react';
import { mount } from 'enzyme';
import FormEditor from '../FormEditor';
import RuntimeToolsFormDetailsContext, {
  FormDetailsContextImpl
} from '../../contexts/FormDetailsContext';
import { act } from 'react-dom/test-utils';
import FormDetailsContextProvider from '../../contexts/FormDetailsContextProvider';

const MockedComponent = (): React.ReactElement => {
  return <></>;
};
jest.mock('@patternfly/react-code-editor/dist/js/components/CodeEditor', () =>
  Object.assign(jest.requireActual('@patternfly/react-code-editor'), {
    CodeEditor: () => {
      return <MockedComponent />;
    },
    Language: {
      json: 'json'
    }
  })
);

const formContent = {
  formInfo: {
    name: 'form1',
    type: 'HTML' as any,
    lastModified: new Date('2020-07-11T18:30:00.000Z')
  },
  source: '<div><span>1</span></div>',
  configuration: {
    resources: {
      styles: {},
      scripts: {}
    },
    schema: 'json schema'
  }
};
const changedValue = '<React.FC><div><span>2</span></div></React.FC>';
const monaco = {
  languages: {
    typescript: {
      typescriptDefaults: {
        setCompilerOptions: jest.fn(),
        setDiagnosticsOptions: jest.fn()
      }
    }
  },
  KeyMod: {
    CtrlCmd: 2048
  },
  KeyCode: {
    49: 'KEY_S'
  }
};

const editor = {
  addCommand: jest.fn(),
  getValue: () => changedValue,
  focus: jest.fn(),
  trigger: jest.fn()
};

describe('FormEditor test', () => {
  it('render source - html', () => {
    const props = {
      code: '<div><span>1</span></div>',
      isSource: true,
      formType: 'html',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };

    const wrapper = mount(<FormEditor {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('render source - tsx', async () => {
    const props = {
      code: '<React.FC><div><span>1</span></div></React.FC>',
      isSource: true,
      formType: 'tsx',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(
      <FormDetailsContextProvider>
        <FormEditor {...props} />
      </FormDetailsContextProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Test execute option - source content', async () => {
    const props = {
      code: '<React.FC><div><span>1</span></div></React.FC>',
      isSource: true,
      formType: 'tsx',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(
      <FormDetailsContextProvider>
        <FormEditor {...props} />
      </FormDetailsContextProvider>
    );
    await act(async () => {
      wrapper.find('CodeEditor').props()['onEditorDidMount'](editor, monaco);
    });
    wrapper.update();
    const childs = wrapper.find('CodeEditor').props()['customControls']
      .props.children;
    childs[0].props.onClick();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('Test execute option - configuration', async () => {
    const props = {
      code: '<React.FC><div><span>1</span></div></React.FC>',
      isSource: false,
      formType: 'tsx',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const changedValue = `{
      "styles": {
        "bootstrap.min.css": " https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
      },
      "scripts": {
        "bootstrap.min.js": "https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js",
        "jquery.js": "https://code.jquery.com/jquery-3.2.1.slim.min.js",
        "popper.js": "https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"
      }
    }`;

    const editorData = {
      addCommand: jest.fn(),
      getValue: () => changedValue,
      focus: jest.fn(),
      trigger: jest.fn()
    };

    const wrapper = mount(
      <FormDetailsContextProvider>
        <FormEditor {...props} />
      </FormDetailsContextProvider>
    );
    await act(async () => {
      wrapper
        .find('CodeEditor')
        .props()
        ['onEditorDidMount'](editorData, monaco);
    });
    wrapper.update();
    const childs = wrapper.find('CodeEditor').props()['customControls']
      .props.children;
    childs[0].props.onClick();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('Test save option', async () => {
    const props = {
      code: '<React.FC><div><span>1</span></div></React.FC>',
      isSource: true,
      formType: 'tsx',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(
      <FormDetailsContextProvider>
        <FormEditor {...props} />
      </FormDetailsContextProvider>
    );
    await act(async () => {
      wrapper.find('CodeEditor').props()['onEditorDidMount'](editor, monaco);
    });
    wrapper.update();
    const childs = wrapper.find('CodeEditor').props()['customControls']
      .props.children;
    childs[3].props.onClick();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });

  it('Test undo option', async () => {
    const props = {
      code: '<React.FC><div><span>1</span></div></React.FC>',
      isSource: true,
      formType: 'tsx',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(
      <FormDetailsContextProvider>
        <FormEditor {...props} />
      </FormDetailsContextProvider>
    );
    await act(async () => {
      wrapper.find('CodeEditor').props()['onEditorDidMount'](editor, monaco);
    });
    wrapper.update();
    const childs = wrapper.find('CodeEditor').props()['customControls']
      .props.children;
    childs[1].props.onClick();
    wrapper.update();
    expect(wrapper.find('CodeEditor').props()['code']).toEqual(
      '<React.FC><div><span>1</span></div></React.FC>'
    );
  });

  it('Test redo option', async () => {
    const props = {
      code: '<React.FC><div><span>1</span></div></React.FC>',
      isSource: true,
      formType: 'tsx',
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(
      <FormDetailsContextProvider>
        <FormEditor {...props} />
      </FormDetailsContextProvider>
    );

    await act(async () => {
      wrapper.find('CodeEditor').props()['onEditorDidMount'](editor, monaco);
    });
    wrapper.update();
    const childs = wrapper.find('CodeEditor').props()['customControls']
      .props.children;
    childs[2].props.onClick();
    wrapper.update();
    expect(wrapper.find('CodeEditor').props()['code']).toEqual(
      '<React.FC><div><span>1</span></div></React.FC>'
    );
  });

  it('render config', () => {
    const props = {
      code: JSON.stringify({
        1: '1',
        2: '2'
      }),
      isConfig: true,
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(<FormEditor {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call refresh', () => {
    const props = {
      code: JSON.stringify({
        1: '1',
        2: '2'
      }),
      isConfig: true,
      formContent: formContent,
      setFormContent: jest.fn(),
      saveFormContent: jest.fn()
    };
    const wrapper = mount(
      <RuntimeToolsFormDetailsContext.Provider
        value={new FormDetailsContextImpl()}
      >
        <FormEditor {...props} />
      </RuntimeToolsFormDetailsContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
