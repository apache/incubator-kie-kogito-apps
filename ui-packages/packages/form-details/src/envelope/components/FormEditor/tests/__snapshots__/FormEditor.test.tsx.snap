// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormEditor test call refresh 1`] = `
<ForwardRef
  code="{\\"1\\":\\"1\\",\\"2\\":\\"2\\"}"
  contentChange={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span><span>2</span></div>",
    }
  }
  formContent={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span></div>",
    }
  }
  isConfig={true}
  saveFormContent={[MockFunction]}
  setContentChange={[MockFunction]}
  setFormContent={[MockFunction]}
>
  <div
    data-ouia-component-type="form-view"
    data-ouia-safe={true}
  >
    <CodeEditor
      code="{\\"1\\":\\"1\\",\\"2\\":\\"2\\"}"
      customControls={
        <React.Fragment>
          <CodeEditorControl
            aria-label="Save form"
            icon={
              <SaveIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Save form"
          />
          <CodeEditorControl
            aria-label="Execute form"
            icon={
              <PlayIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Execute form"
          />
          <CodeEditorControl
            aria-label="Undo changes"
            icon={
              <UndoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Undo changes"
          />
          <CodeEditorControl
            aria-label="Redo changes"
            icon={
              <RedoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Redo changes"
          />
        </React.Fragment>
      }
      height="800px"
      isCopyEnabled={true}
      isDarkTheme={false}
      isLanguageLabelVisible={true}
      isLineNumbersVisible={true}
      isMinimapVisible={false}
      isReadOnly={false}
      language="json"
      onChange={[Function]}
      onEditorDidMount={[Function]}
    >
      <MockedComponent />
    </CodeEditor>
  </div>
</ForwardRef>
`;

exports[`FormEditor test render config 1`] = `
<ForwardRef
  code="{\\"1\\":\\"1\\",\\"2\\":\\"2\\"}"
  contentChange={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span><span>2</span></div>",
    }
  }
  formContent={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span></div>",
    }
  }
  isConfig={true}
  saveFormContent={[MockFunction]}
  setContentChange={[MockFunction]}
  setFormContent={[MockFunction]}
>
  <div
    data-ouia-component-type="form-view"
    data-ouia-safe={true}
  >
    <CodeEditor
      code="{\\"1\\":\\"1\\",\\"2\\":\\"2\\"}"
      customControls={
        <React.Fragment>
          <CodeEditorControl
            aria-label="Save form"
            icon={
              <SaveIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Save form"
          />
          <CodeEditorControl
            aria-label="Execute form"
            icon={
              <PlayIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Execute form"
          />
          <CodeEditorControl
            aria-label="Undo changes"
            icon={
              <UndoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Undo changes"
          />
          <CodeEditorControl
            aria-label="Redo changes"
            icon={
              <RedoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Redo changes"
          />
        </React.Fragment>
      }
      height="800px"
      isCopyEnabled={true}
      isDarkTheme={false}
      isLanguageLabelVisible={true}
      isLineNumbersVisible={true}
      isMinimapVisible={false}
      isReadOnly={false}
      language="json"
      onChange={[Function]}
      onEditorDidMount={[Function]}
    >
      <MockedComponent />
    </CodeEditor>
  </div>
</ForwardRef>
`;

exports[`FormEditor test render source - html 1`] = `
<ForwardRef
  code="<div><span>1</span></div>"
  contentChange={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span><span>2</span></div>",
    }
  }
  formContent={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span></div>",
    }
  }
  formType="html"
  isSource={true}
  saveFormContent={[MockFunction]}
  setContentChange={[MockFunction]}
  setFormContent={[MockFunction]}
>
  <div
    data-ouia-component-type="form-view"
    data-ouia-safe={true}
  >
    <CodeEditor
      code="<div><span>1</span></div>"
      customControls={
        <React.Fragment>
          <CodeEditorControl
            aria-label="Save form"
            icon={
              <SaveIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Save form"
          />
          <CodeEditorControl
            aria-label="Execute form"
            icon={
              <PlayIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Execute form"
          />
          <CodeEditorControl
            aria-label="Undo changes"
            icon={
              <UndoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Undo changes"
          />
          <CodeEditorControl
            aria-label="Redo changes"
            icon={
              <RedoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Redo changes"
          />
        </React.Fragment>
      }
      height="800px"
      isCopyEnabled={true}
      isDarkTheme={false}
      isLanguageLabelVisible={true}
      isLineNumbersVisible={true}
      isMinimapVisible={false}
      isReadOnly={false}
      language="html"
      onChange={[Function]}
      onEditorDidMount={[Function]}
    >
      <MockedComponent />
    </CodeEditor>
  </div>
</ForwardRef>
`;

exports[`FormEditor test render source - tsx 1`] = `
<ForwardRef
  code="<React.FC><div><span>1</span></div></React.FC>"
  contentChange={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span><span>2</span></div>",
    }
  }
  formContent={
    Object {
      "configuration": Object {
        "resources": Object {
          "scripts": Object {},
          "styles": Object {},
        },
        "schema": "json schema",
      },
      "formInfo": Object {
        "lastModified": 2020-07-11T18:30:00.000Z,
        "name": "form1",
        "type": "HTML",
      },
      "source": "<div><span>1</span></div>",
    }
  }
  formType="tsx"
  isSource={true}
  saveFormContent={[MockFunction]}
  setContentChange={[MockFunction]}
  setFormContent={[MockFunction]}
>
  <div
    data-ouia-component-type="form-view"
    data-ouia-safe={true}
  >
    <CodeEditor
      code="<React.FC><div><span>1</span></div></React.FC>"
      customControls={
        <React.Fragment>
          <CodeEditorControl
            aria-label="Save form"
            icon={
              <SaveIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Save form"
          />
          <CodeEditorControl
            aria-label="Execute form"
            icon={
              <PlayIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Execute form"
          />
          <CodeEditorControl
            aria-label="Undo changes"
            icon={
              <UndoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Undo changes"
          />
          <CodeEditorControl
            aria-label="Redo changes"
            icon={
              <RedoIcon
                className="pf-global--primary-color--100"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            }
            isVisible={true}
            onClick={[Function]}
            toolTipText="Redo changes"
          />
        </React.Fragment>
      }
      height="800px"
      isCopyEnabled={true}
      isDarkTheme={false}
      isLanguageLabelVisible={true}
      isLineNumbersVisible={true}
      isMinimapVisible={false}
      isReadOnly={false}
      language="typescript"
      onChange={[Function]}
      onEditorDidMount={[Function]}
    >
      <MockedComponent />
    </CodeEditor>
  </div>
</ForwardRef>
`;
